SpinCore_pp/ppg/echo.py

```diff
@@ -86,14 +89,16 @@ def run_spin_echo(
         phase steps for the second pulse
     ret_data: nddata (default None)
         returned data from previous run or `None` for the first run.
+    plen_as_beta: boolean
+        Is plen supplied as a β value [s√W] or directly as programmed length (μs)
     """
     assert nEchoes == 1, "you must only choose nEchoes=1"
     # take the desired p90 and p180
     # (2*desired_p90) and convert to what needs to
     # be programmed in order to get the desired
     # times
-    prog_p90_us = prog_plen(p90_us)
-    prog_p180_us = prog_plen(2 * p90_us)
+    prog_p90_us = prog_plen(plen, amplitude) if plen_as_beta else plen
+    prog_p180_us = prog_plen(2 * plen, amplitude) if plen_as_beta else plen
```

Comment:
please see what I'm doing here, and do the same for the other files in the ppg directory